.github/workflows/publish-charms.yaml

```diff
@@ -30,8 +30,17 @@ jobs:
             echo "::error Failed to determine track/risk from branch ${BRANCH}"
             exit 1
           fi
+  charmcraft-channel:
+    runs-on: ubuntu-latest
+    outputs:
+      channel: ${{ steps.charmcraft.outputs.channel }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Read charmcraft version file
+        id: charmcraft
+        run: echo "channel=$(cat .charmcraft-channel)" >> $GITHUB_OUTPUT
```

Comment:
where does `.charmcraft-channel` comes from?

Reply:
it's a file per branch which signals to the CI which `charmcraft` snap channel to use when building

https://github.com/canonical/k8s-operator/blob/main/.charmcraft-channel

Reply:
cool, thanks!